src/libmw/libmw_node.cpp

```diff
@@ -98,6 +99,25 @@ MWEXPORT void CheckTransaction(const libmw::TxRef& transaction)
     BlockSumValidator::ValidateForTx(*transaction.pTransaction);
 }
 
+MWEXPORT void CheckTxInputs(const libmw::TxRef& transaction, const libmw::CoinsViewRef& view, int nSpendHeight)
+{
+    assert(transaction.pTransaction != nullptr);
+    assert(view.pCoinsView != nullptr);
+
+    auto pCoinsView = std::dynamic_pointer_cast<mw::CoinsViewCache>(view.pCoinsView);
```

Comment:
GetUTXOs is the only API you call, and that's on the base ICoinsView interface. No need for this cast and assertion